pkg/util/crypto.go

```diff
@@ -97,3 +98,67 @@ func GenerateSelfSignedCert(host, certPath, keyPath string) error {
 
 	return nil
 }
+
+// CertPoolFromFile returns an x509.CertPool containing the certificates in the given PEM-encoded file.
+// Returns an error if the file could not be read, a certificate could not be parsed, or if the file does not contain any certificates
+func CertPoolFromFile(filename string) (*x509.CertPool, error) {
```

Comment:
Right now only this function is used outside the util package. Why not restrict the visibility of the helper functions until it's necessary for outside code to call them directly?